File: TreeSortTest/TreeSortTest/NodeT.cs
namespace TreeSortTest;

public class NodeT(int value)
{
               private int _value = value;

               public int Value
               {
                              get => _value;
                              set => this._value = value;
               }

               public NodeT? Left { get; set; }

               public NodeT? Right { get; set; }

               public override String ToString()
               {
                              return value.ToString();
               }
}

File: tree_sort_a_tree-4SB2-BabjarcikD/NodeT.java

public class NodeT {
    private int value;
    private NodeT left;
    private NodeT right;

    public NodeT(int value){
        this.value = value;
        left = null;
        right = null;
    }

    public int getValue() {
        return value;
    }

    public NodeT getLeft() {
        return left;
    }

    public NodeT getRight() {
        return right;
    }

    public void setValue(int value) {
        this.value = value;
    }

    public void setLeft(NodeT left) {
        this.left = left;
    }

    public void setRight(NodeT right) {
        this.right = right;
    }

    @Override
    public String toString() {
        return "NodeT{" +
                "value=" + value +
                ", left=" + left +
                ", right=" + right +
                '}';
    }
}
